File: AILang/benchmarks/python/fixed_array_sum.py
def fixed_array_sum_bench(iterations: int) -> int:
    arr = [3, 1, 4, 1, 5, 9, 2, 6]
    acc = 0
    i = 0
    while i < iterations:
        j = 0
        while j < 8:
            acc += arr[j]
            j += 1
        i += 1
    return acc


def main() -> None:
    iterations = 250000
    result = fixed_array_sum_bench(iterations)
    print(result)


if __name__ == "__main__":
    main()

File: AILang/benchmarks/rust/fixed_array_sum.rs
fn fixed_array_sum_bench(iterations: i64) -> i64 {
    let arr: [i64; 8] = [3, 1, 4, 1, 5, 9, 2, 6];
    let mut acc: i64 = 0;
    for _ in 0..iterations {
        for j in 0..8 {
            acc += arr[j as usize];
        }
    }
    acc
}

fn main() {
    let iterations: i64 = 250_000;
    let result = fixed_array_sum_bench(iterations);
    println!("{result}");
}
